File: javascript/angular2/components/button/src/main.component.js
(function(ButtonModule) {
	"use strict";
	ButtonModule.MainComponent = ng.core
		.Component({
			selector: 'main',
			templateUrl: 'src/buttons-template.html',
			directives: [ButtonModule.ButtonComponent]
		})
		.Class({
			constructor: function() {
				this.info = {value: 'key'};
			},
			procedure: function(data) {
				console.log(data);
			}
		});
})(window.ButtonModule || (window.ButtonModule = {}));

File: dart/angular2/lib/components/button/main_component.dart
library angular2_dart_examples.main_component;

import 'dart:html';
import 'package:angular2/angular2.dart';
import 'package:angular2_dart_examples/components/button/button_component.dart';

@Component(
	selector: 'main',
	templateUrl: 'buttons-template.html',
	directives: const [ButtonComponent])
class MainComponent {
	Map<String, String> info;
	
	MainComponent() {
		// Example property
		info = {'key': 'value'};
	}
	
	// Example method
	dynamic procedure(dynamic data) {
		window.console.log(data);
	}
}